src/shared/services/json-store.service.spec.ts

```diff
@@ -86,4 +86,79 @@ describe('JsonStoreService', () => {
     service.setIn(path, value);
   });
 
+  it('should shift the first element if the new pushed element exceeds the maximum history length', () => {
```

Comment:
since the variables like `history` and `limit` are gonna be `private`, to test this:

* `setJson(initialJson)`
* `addHistory()`
* `setJson(anotherJson)`
* `addHistory()` x <limit> times
* `rollbackHistory()` x <limit> times (depending on the other comment you might need to change the json before each rollback)
*   `subscribe` and expect `changedJson` not equal to `initialJson`